### PR TITLE
[Patient Intake Sample] [Part 2] Add emergency contact, allergies, and medications to the questionnaire

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Partial work #4999 - Add the following fields to the patient intake questionnaire:
  - Emergency contact
  - Allergies 
  - Current medications

## Steps to test

- [ ] Upload core data
- [ ] Check if the changes are reflected in the questionnaire

[Screencast from 06-08-2024 16:35:56.webm](https://github.com/user-attachments/assets/45625501-d539-4c3f-b68a-4c85675282ab)

### CODE REVIEW

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 1]
```diff
@@ -392,52 +466,38 @@ export function getGroupRepeatedAnswers(
...
-  let linkCursor = 0;
-
-  const groupAnswers = [];
-  let answerGroup = {};
+  if (!responseGroups || responseGroups?.length === 0) {
+    return [];
+  }
 
-  // It expects that responses will be organized in the same order the form is filled. Eg.:
+  // It expects that responses will be organized in separate groups with the same groupLinkId. Eg.:
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:483]: The `getGroupRepeatedAnswers` helper was considering a wrong questionnaire response structure, I fixed it and its related code in the `intake-form-test-data.ts` file.

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: Please let me know if there are any text copy changes to make here.

File: examples/medplum-patient-intake-demo/src/bots/core/test-data/intake-form-test-data.ts [hunk 2]
```diff
@@ -201,6 +201,178 @@ export const intakeResponse: QuestionnaireResponse = {
         },
       ],
     },
+    {
+      id: 'id-82',
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/test-data/intake-form-test-data.ts:205]: Is it okay to remove those `id`s from the test data, or is it a good pattern to have them?

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json
Comment (reshmakh) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: @pamella - remove all references to "BACH" and replace with "Sample Hospital"

Comment (reshmakh) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: Also, on the Side Nav - if you could add a section called "Onboarding" and an item "New Patient"
image

Comment (pamella) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: > remove all references to "BACH" and replace with "Sample Hospital"

Those references have already been removed at #5026

Comment (pamella) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: > Also, on the Side Nav - if you could add a section called "Onboarding" and an item "New Patient" image

Of course. I had created the issue #5015 to track that request. My intention is to work on it after I finish the work-in-progress on issue #4999, please let me know if this works.

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 3]
```diff
@@ -295,6 +295,80 @@ export function addLanguage(patient: Patient, valueCoding: Coding | undefined, p
...
+
+  if (!code) {
+    return;
+  }
+
+  const note = answers['medication-note']?.valueString;
+
+  await medplum.upsertResource(
+    {
+      resourceType: 'MedicationStatement',
```
[the remark this answers is not in the document]
Comment (reshmakh) [reply examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:359]: That's correct, but it does not show up in the PatientSummary.  Filed an issue to track adding MedicationStatement to patient summary. https://github.com/medplum/medplum/issues/5036

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json
Comment (reshmakh) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: Line 378 still has BACH references

Comment (pamella) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: Sorry about that. It's fixed now

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json [hunk 4]
```diff
@@ -142,6 +142,83 @@
...
+                "linkId": "allergy-reaction",
+                "text": "Reaction",
+                "type": "string"
+              },
+              {
+                "linkId": "allergy-clinical-status",
+                "text": "Clinical Status",
+                "type": "choice",
+                "answerValueSet": "http://hl7.org/fhir/ValueSet/allergyintolerance-clinical"
+              },
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:195]: We actually can omit this, and default anything that is entered to "active"

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 5]
```diff
@@ -295,6 +295,82 @@ export function addLanguage(patient: Patient, valueCoding: Coding | undefined, p
...
+  const clinicalStatus = answers['allergy-clinical-status']?.valueCoding;
+  const reaction = answers['allergy-reaction']?.valueString;
+  const onsetDateTime = answers['allergy-onset']?.valueDateTime;
+
+  await medplum.upsertResource(
+    {
+      resourceType: 'AllergyIntolerance',
+      patient: createReference(patient),
+      code: { coding: [code] },
+      clinicalStatus: clinicalStatus ? { coding: [clinicalStatus] } : undefined,
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:325]: Should also default the `verificationStatus` to `unconfirmed`

File: examples/medplum-patient-intake-demo/src/bots/core/test-data/intake-form-test-data.ts (hunk 2 above)
Comment (rahul1) [reply examples/medplum-patient-intake-demo/src/bots/core/test-data/intake-form-test-data.ts:205]: ok to remove!

**Review by rahul1 — APPROVED** [review]
@pamella a few minor comments about allergy statuses.

**Comment by pamella** [comment]
@rahul1 Thanks! I'll fix that in the PR https://github.com/medplum/medplum/pull/5059

**Pull request merged**